File: examples/nodejs/simple_bot.js
/**
 * Simple Node.js bot example for Sapine Bot Hosting Platform.
 * This bot prints messages to demonstrate the logging system.
 */

const formatTime = () => {
    const now = new Date();
    return now.toTimeString().split(' ')[0];
};

const main = async () => {
    console.log("=".repeat(50));
    console.log("Simple Node.js Bot");
    console.log("=".repeat(50));
    console.log(`Started at: ${new Date().toISOString()}`);
    console.log();

    let counter = 0;

    const interval = setInterval(() => {
        counter++;
        console.log(`[${formatTime()}] Bot is running... (iteration ${counter})`);

        // Log every 10 iterations
        if (counter % 10 === 0) {
            console.log(`✓ Completed ${counter} iterations`);
        }
    }, 5000);

    // Handle graceful shutdown
    process.on('SIGINT', () => {
        clearInterval(interval);
        console.log("\n" + "=".repeat(50));
        console.log("Bot stopped by user");
        console.log(`Total iterations: ${counter}`);
        console.log("=".repeat(50));
        process.exit(0);
    });

    process.on('SIGTERM', () => {
        clearInterval(interval);
        console.log("\n" + "=".repeat(50));
        console.log("Bot stopped");
        console.log(`Total iterations: ${counter}`);
        console.log("=".repeat(50));
        process.exit(0);
    });
};

main().catch(error => {
    console.error("Fatal error:", error);
    process.exit(1);
});

File: examples/python/simple_bot.py
"""
Simple Python bot example for Sapine Bot Hosting Platform.
This bot prints messages to demonstrate the logging system.
"""

import time
from datetime import datetime

def main():
    print("=" * 50)
    print("Simple Python Bot")
    print("=" * 50)
    print(f"Started at: {datetime.now().isoformat()}")
    print()
    
    counter = 0
    try:
        while True:
            counter += 1
            print(f"[{datetime.now().strftime('%H:%M:%S')}] Bot is running... (iteration {counter})")
            
            # Do some work
            time.sleep(5)
            
            # Log every 10 iterations
            if counter % 10 == 0:
                print(f"✓ Completed {counter} iterations")
    
    except KeyboardInterrupt:
        print("\n" + "=" * 50)
        print("Bot stopped by user")
        print(f"Total iterations: {counter}")
        print("=" * 50)

if __name__ == "__main__":
    main()
